File: app/src/main/kotlin/com/giulia/diversification/data/model/FoodEntry.kt
package com.giulia.diversification.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "food_entries",
    foreignKeys = [ForeignKey(
        entity = Food::class,
        parentColumns = ["id"],
        childColumns = ["foodId"],
        onDelete = ForeignKey.CASCADE
    )],
    indices = [Index("foodId")]
)
data class FoodEntry(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val foodId: Int,
    val dateIntroduced: Long = System.currentTimeMillis(),
    val reaction: Reaction = Reaction.NEUTRAL,
    val hasDigestiveIssue: Boolean = false,
    val hasAllergicReaction: Boolean = false,
    val notes: String? = null
)

File: ios/PetitesBouchees/Models/FoodEntry.swift
import SwiftData
import Foundation

@Model
final class FoodEntry {
    var dateIntroduced: Date = Date.now
    var reactionRawValue: String = "neutral"
    var hasDigestiveIssue: Bool = false
    var hasAllergicReaction: Bool = false
    var notes: String?
    var food: Food?

    var reaction: Reaction {
        get { Reaction(rawValue: reactionRawValue) ?? .neutral }
        set { reactionRawValue = newValue.rawValue }
    }

    init(food: Food, dateIntroduced: Date = .now, reaction: Reaction = .neutral, hasDigestiveIssue: Bool = false, hasAllergicReaction: Bool = false, notes: String? = nil) {
        self.food = food
        self.dateIntroduced = dateIntroduced
        self.reactionRawValue = reaction.rawValue
        self.hasDigestiveIssue = hasDigestiveIssue
        self.hasAllergicReaction = hasAllergicReaction
        self.notes = notes
    }
}
